### PR TITLE
Add page_view model

### PR DESCRIPTION
**Pull request by benhalpern**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description
This feature adds custom tracking of page views from within the app to complement and compare against Google Analytics data. This should make initial reporting faster and more reliable.

This also opens up the opportunity to show users their history.

### CODE REVIEW

File: app/assets/javascripts/initializers/initializeBaseTracking.js.erb [hunk 1]
```diff
@@ -116,6 +116,29 @@ function trackCustomImpressions() {
         successLinks[i].addEventListener('click', function() { trackHtmlVariantSuccess(dataBody, csrfToken) });
       }
     }
+
+    // page view
+    if (ArticleElement && tokenMeta && !isBot) {
+      var randomNumber = Math.floor(Math.random() * 10); // 1 in 10; Only track 1 in 10 impressions
```
Comment (Glennmen) [inline app/assets/javascripts/initializers/initializeBaseTracking.js.erb]: Why don't you want to track all views?

Comment (lightalloy) [reply app/assets/javascripts/initializers/initializeBaseTracking.js.erb]: I suppose that it's done for optimization, to make fewer requests and db writes. For unauthorized users, only one of each 10 requests will be saved, but the count will be 10 (https://github.com/thepracticaldev/dev.to/pull/1985/files#diff-a71fed5e724b3cd9cce62138dfb4d631R10), so the number of views will remain approximately the same as if every request was tracked.

File: app/assets/javascripts/initializers/initializeBaseTracking.js.erb [hunk 2]
```diff
@@ -116,6 +116,29 @@ function trackCustomImpressions() {
...
+      if (!checkUserLoggedIn() && randomNumber != 1) {
```
Comment (lightalloy) [inline app/assets/javascripts/initializers/initializeBaseTracking.js.erb]: I suggest transforming condition to smth like `if !(checkUserLoggedIn() || randomNumber == 1)`, in my opinion it's easier to understand.

File: app/assets/javascripts/initializers/initializeBaseTracking.js.erb (hunk 1 above)
Comment (benhalpern) [reply app/assets/javascripts/initializers/initializeBaseTracking.js.erb]: Yup, that’s the logic. We can adjust things later, but this should be a bit less resource costly and give us good enough insights.

**Pull request merged**